pulp_rpm/app/modulemd.py

```diff
@@ -50,130 +58,121 @@ def modules_packages(modules):
     version.add_content(Package.objects.filter(pk__in=packages_to_add))
 
 
-def parse_modulemd(module_names, module_index):
+def split_modulmd_file(file):
     """
-    Get modulemd NSVCA, artifacts, dependencies.
-
-    Args:
-        module_names (list):
-            list of modulemd names
-        module_index (mmdlib.ModuleIndex):
-            libmodulemd index object
+    Helper method to preserve original formatting of modulemd.
+    """
+    m_open = gzip.open if file.url.endswith(".gz") else open
```

Comment:
Careful, we [ran into this problem before](https://github.com/pulp/pulp_rpm/pull/1983/).  CentOS uses [XZ compression](http://mirror.centos.org/centos/8-stream/AppStream/x86_64/os/repodata/).  We need to support the same types that libmodulemd supports.